### PR TITLE
012 fetch all grades

### LINKED ISSUE #12: GET /assignments/grades/:assignmentID -> Fetch all grades for specified assignment
Request
-null
Response
-status
-grades

### PR DESCRIPTION
**Pull request by GuruprasadChoukimath**
Please merge #34  first

### CODE REVIEW

File: backend/data/users.js [hunk 1]
```diff
@@ -301,6 +301,27 @@ async function addGrade(studentId, courseId, assignmentId, grade) {
...
+/**
+ * Fetch grade of a student by assignmentId
+ * @param {string} assignmentId 
+ * @param {string} studentId 
+ * @returns object
+ */
+async function fetchGrade(assignmentId, studentId) {
+  const errorMSg = 'No assignments found for the given assignment id';
+  const user = await getUser(studentId);
+  if (user && !user.classes) throw new Error(errorMSg);
```
Comment (karl-olsen) [inline backend/data/users.js:313]: Might be good to add a check for if(!user) with a throw, just to cover all bases

Comment (GuruprasadChoukimath) [reply backend/data/users.js:313]: Thanks, @karl-olsen I have added the changes in the latest PR

File: backend/routes/assignments.js [hunk 2]
```diff
@@ -147,4 +147,29 @@ router.get('/grades/:assignmentId', auth, async (req, res) => {
 
 })
 
+/**
+ * Fetch all grades for given assignment
+ * Route: GET
+ * Params: assignmentId
+ */
+router.get('/grades/all/:assignmentId', auth, async (req, res) => {
```
Comment (karl-olsen) [inline backend/routes/assignments.js:155]: Not that this is necessarily dealbreaking, but specs say the route should just be:
/assignments/grade/:assignmentId
Is there a reason for the /all that I might just not be aware of?

Comment (GuruprasadChoukimath) [reply backend/routes/assignments.js:155]: Yes, there is a conflicting route with the same pattern. Due to the same route pattern, the route in #13 will interpret the request first and the request never comes to the above method. Hence I had to change the path

**Pull request merged**